### PR TITLE
Added flows.update_flow(...)

### PR DESCRIPTION
**Pull request by derek-globus**
[sc-22685](https://app.shortcut.com/globus/story/22685)
Added the `update_flow` method to `FlowsClient`

Direct Generated Docs Link: [FlowsClient.update_flow()](https://globus-sdk-python--710.org.readthedocs.build/en/710/services/flows.html#globus_sdk.FlowsClient.update_flow)

## Related Changes
I additionally modified the `create_flow` docs to be slightly shorter (by using [sphinx-design dropdowns](https://sphinx-design.readthedocs.io/en/furo-theme/dropdowns.html)) and pulled the example usage into a tabbed view along with an example response
  * This change felt small & related enough to `update_flow` but I can rip it out of this PR if anyone would prefer I submit it separately

## Testing
* Hit flows with various normal editing permutations of flow definition, input_schema, flow_viewers, etc.
* Tested the flow_owner update (this one felt fairly special-cased so I wanted to verify it specifically)
   * Created a flow with user A with user B as flow_administrator
   * As user B, called `flows.update_flow(flow_id, flow_owner=user_b_urn)
   * Observed user B now owned the Flow

----
:books: Documentation preview :books:: https://globus-sdk-python--710.org.readthedocs.build/en/710/

### CODE REVIEW

File: src/globus_sdk/services/flows/client.py [hunk 1]
```diff
@@ -270,6 +276,143 @@ def list_flows(
...
+        Any fields omitted from the request will be unchanged
+
+        :param flow_id: The ID of the Flow to fetch
+        :type flow_id: str or UUID
+        :param title: A non-unique, human-friendly name used for displaying the
+            flow to end users.
+        :type title: str (1 - 128 characters), optional
+        :param definition: JSON object specifying flows states and execution order. For
+            a more detailed explanation of the flow definition, see
+            `Authoring Flows <https://globus-automate-client.readthedocs.io/en/latest/authoring_flows.html>`_
```
Comment (kurtmckee) [inline src/globus_sdk/services/flows/client.py:308]: I think this URL is incorrect.

File: src/globus_sdk/services/flows/client.py [hunk 2]
```diff
@@ -270,6 +276,143 @@ def list_flows(
...
+        :type definition: dict, optional
+        :param input_schema: A JSON Schema to which Flow Invocation input must conform
+        :type input_schema: dict, optional
+        :param subtitle: A concise summary of the flow’s purpose.
+        :type subtitle: str (0 - 128 characters), optional
+        :param description: A detailed description of the flow's purpose for end user
+            display.
+        :type description: str (0 - 4096 characters), optional
+        :param flow_owner: An Auth Identity URN to set as flow owner; this must match
+            the Identity URN of the entity calling `update_flow`
```
Comment (kurtmckee) [inline src/globus_sdk/services/flows/client.py:318]: "the" identity URN, or "an" identity URN of any of the caller's linked accounts?

File: src/globus_sdk/services/flows/client.py [hunk 3]
```diff
@@ -270,6 +276,143 @@ def list_flows(
...
+        :type flow_owner: str, optional
+        :param flow_viewers: A set of Principal URN values, or the value "public"
+            indicating entities who can view the flow
```
Comment (kurtmckee) [inline src/globus_sdk/services/flows/client.py:321]:
```suggestion
        :param flow_viewers: A set of Principal URN values, or the value "public",
            indicating entities who can view the flow
```

File: src/globus_sdk/services/flows/client.py [hunk 4]
```diff
@@ -270,6 +276,143 @@ def list_flows(
...
+                .. code-block:: json
+
+                    [ "public" ]
+
+                .. code-block:: json
+
+                    [
+                        "urn:globus:auth:identity:b44bddda-d274-11e5-978a-9f15789a8150",
+                        "urn:globus:groups:id:c1dcd951-3f35-4ea3-9f28-a7cdeaf8b68f"
+                    ]
```
Comment (kurtmckee) [inline src/globus_sdk/services/flows/client.py:334]: LOVE. THIS. :heart:

File: src/globus_sdk/services/flows/client.py [hunk 5]
```diff
@@ -270,6 +276,143 @@ def list_flows(
...
+
+        :type flow_viewers: list[str], optional
+        :param flow_starters: A set of Principal URN values, or the value
+            "all_authenticated_users" indicating entities who can initiate a *Run* of
+            the flow
```
Comment (kurtmckee) [inline src/globus_sdk/services/flows/client.py:339]:
```suggestion
        :param flow_starters: A set of Principal URN values, or the value
            "all_authenticated_users", indicating entities who can initiate a *Run* of
            the flow
```

File: src/globus_sdk/services/flows/client.py [hunk 6]
```diff
@@ -270,6 +276,143 @@ def list_flows(
...
+                        "urn:globus:auth:identity:b44bddda-d274-11e5-978a-9f15789a8150",
+                        "urn:globus:groups:id:c1dcd951-3f35-4ea3-9f28-a7cdeaf8b68f"
+                    ]
+
+        :type flow_administrators: list[str], optional
+        :param keywords: A set of terms used to categorize the flow used in query and
+            discovery operations
+        :type keywords: list[str] (0 - 1024 items), optional
+        :param additional_fields: Additional Key/Value pairs sent to the create API
+        :type additional_fields: dict or str -> any, optional
```
Comment (kurtmckee) [inline src/globus_sdk/services/flows/client.py:374]: This wording confuses me. Should this be "dict _of_ str -> any"?

```suggestion
        :type additional_fields: dict of str -> any, optional
```

**Review by kurtmckee — CHANGES_REQUESTED** [review]

File: src/globus_sdk/services/flows/client.py (hunk 2 above)
Comment (derek-globus) [reply src/globus_sdk/services/flows/client.py:318]: Hadn't considered this.
Lemme verify that it is in fact any identity associated with the new-owner account.

Good callout

File: src/globus_sdk/services/flows/client.py (hunk 1 above)
Comment (derek-globus) [reply src/globus_sdk/services/flows/client.py:308]: Updated

File: src/globus_sdk/services/flows/client.py (hunk 2 above)
Comment (derek-globus) [reply src/globus_sdk/services/flows/client.py:318]: Did some testing, `"the Identity URN"` is correct.

Linked accounts are not allowed to set each other as the owner of a flow.

---

Given the following:
3 identities:
* A-1
* A-2
* B

where A-1 & A-2 are linked identities

I did the following:
1. As B: Create a flow (B is implicitly owner)
2. As B: Add A-1, A-2, B as Administrators of the flow
3. As A-1, attempt to set A-2 as FlowOwner -> Error 400
   * ```
      "Invalid value for field 'owner': if present it must match the identity of the calling user (<A-1's URN>) and that user must be an Administrator of the Flow"
      ```
4. As A-1, attempt to set A-1 as Flow Owner -> success
5. As A-1, attempt again to set A-2 as Flow Owner -> Same 400 as in step (3)

**Review by sirosen — APPROVED** [review]

**Review by kurtmckee — APPROVED** [review]

**Pull request merged**